openquake/hazardlib/tests/site_amplification_test.py

```diff
@@ -102,6 +102,20 @@ class AmplifierTestCase(unittest.TestCase):
               [.999, .995, .99, .98, .95, .9, .8, .7, .1, .05, .01],  # SA(0.2)
               [.999, .995, .99, .98, .95, .9, .8, .7, .1, .05, .01]]  # SA(0.5)
 
+    def test_missing_defined_for_reference_velocity(self):
+        fname = gettemp(trivial_ampl_func)
+        df = read_csv(fname, {'ampcode': ampcode_dt, None: numpy.float64},
+                      index='ampcode')
+        a = Amplifier(self.imtls, df, self.soil_levels)
+        gmm = valid.gsim('CanadaSHM6_ActiveCrust_BooreEtAl2014')
+        with self.assertRaises(AttributeError) as ctx:
+            a.check(self.vs30, 0, {TRT.ACTIVE_SHALLOW_CRUST: [gmm]})
+        self.assertIn('The parameter "vs30_tolerance" is >= 0, but'
+                      ' the attribute DEFINED_FOR_REFERENCE_VELOCITY'
+                      ' is missing in the gsim'
+                      ' [CanadaSHM6_ActiveCrust_BooreEtAl2014]',
+                      str(ctx.exception))
```

Comment:
You also need a test with CanadaSHM6_ActiveCrust_BooreEtAl2014 and vs30_tolerance=-1.